<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>MainWindow</class>
 <widget class="QMainWindow" name="MainWindow">
  <property name="windowModality">
   <enum>Qt::NonModal</enum>
  </property>
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1000</width>
    <height>649</height>
   </rect>
  </property>
  <property name="sizePolicy">
   <sizepolicy hsizetype="Fixed" vsizetype="Fixed">
    <horstretch>0</horstretch>
    <verstretch>0</verstretch>
   </sizepolicy>
  </property>
  <property name="maximumSize">
   <size>
    <width>16777215</width>
    <height>16777215</height>
   </size>
  </property>
  <property name="contextMenuPolicy">
   <enum>Qt::ActionsContextMenu</enum>
  </property>
  <property name="windowTitle">
   <string>SDN Topology</string>
  </property>
  <property name="windowIcon">
   <iconset resource="../images/img.qrc">
    <normaloff>:/images/controller.png</normaloff>:/images/controller.png</iconset>
  </property>
  <property name="locale">
   <locale language="English" country="UnitedStates"/>
  </property>
  <widget class="QWidget" name="centralWidget">
   <layout class="QGridLayout" name="gridLayout">
    <item row="0" column="0">
     <widget class="QScrollArea" name="scrollArea">
      <property name="sizePolicy">
       <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
        <horstretch>0</horstretch>
        <verstretch>0</verstretch>
       </sizepolicy>
      </property>
      <property name="widgetResizable">
       <bool>true</bool>
      </property>
      <widget class="QWidget" name="scrollContents">
       <property name="geometry">
        <rect>
         <x>0</x>
         <y>0</y>
         <width>1366</width>
         <height>1024</height>
        </rect>
       </property>
       <layout class="QGridLayout" name="gridLayout_2">
        <property name="leftMargin">
         <number>0</number>
        </property>
        <property name="topMargin">
         <number>0</number>
        </property>
        <property name="rightMargin">
         <number>0</number>
        </property>
        <property name="bottomMargin">
         <number>0</number>
        </property>
        <item row="0" column="0">
         <widget class="NetworkMapView" name="networkMapView">
          <property name="sizePolicy">
           <sizepolicy hsizetype="Preferred" vsizetype="Preferred">
            <horstretch>0</horstretch>
            <verstretch>0</verstretch>
           </sizepolicy>
          </property>
          <property name="minimumSize">
           <size>
            <width>1366</width>
            <height>1024</height>
           </size>
          </property>
          <property name="mouseTracking">
           <bool>false</bool>
          </property>
          <property name="contextMenuPolicy">
           <enum>Qt::NoContextMenu</enum>
          </property>
          <property name="text">
           <string/>
          </property>
          <property name="textInteractionFlags">
           <set>Qt::NoTextInteraction</set>
          </property>
         </widget>
        </item>
       </layout>
      </widget>
     </widget>
    </item>
    <item row="1" column="0">
     <widget class="QTextBrowser" name="terminal">
      <property name="maximumSize">
       <size>
        <width>16777215</width>
        <height>200</height>
       </size>
      </property>
     </widget>
    </item>
   </layout>
  </widget>
  <widget class="QMenuBar" name="menuBar">
   <property name="geometry">
    <rect>
     <x>0</x>
     <y>0</y>
     <width>1000</width>
     <height>21</height>
    </rect>
   </property>
   <widget class="QMenu" name="menuFile">
    <property name="title">
     <string>File</string>
    </property>
    <addaction name="actionNew"/>
    <addaction name="actionOpen"/>
    <addaction name="actionSave"/>
    <addaction name="actionSaveAs"/>
    <addaction name="separator"/>
    <addaction name="actionExit"/>
   </widget>
   <widget class="QMenu" name="menuConverter">
    <property name="title">
     <string>Export</string>
    </property>
    <addaction name="actionCreateMnDataFile"/>
    <addaction name="actionWeights_Matrix"/>
   </widget>
   <widget class="QMenu" name="menuOptions">
    <property name="title">
     <string>Options</string>
    </property>
    <widget class="QMenu" name="menuShow_metric">
     <property name="title">
      <string>Show metric</string>
     </property>
     <addaction name="actionDisplayDelay"/>
     <addaction name="actionDisplayBandwidth"/>
     <addaction name="actionDisplayPacketLoss"/>
    </widget>
    <addaction name="actionShowPorts"/>
    <addaction name="menuShow_metric"/>
    <addaction name="actionRandomMetric"/>
   </widget>
   <widget class="QMenu" name="menuInstruments">
    <property name="title">
     <string>Tools</string>
    </property>
    <addaction name="actionSdnController"/>
    <addaction name="actionSwitch"/>
    <addaction name="actionHost"/>
    <addaction name="actionLink"/>
    <addaction name="actionText"/>
    <addaction name="actionEdit"/>
   </widget>
   <widget class="QMenu" name="menuActions">
    <property name="title">
     <string>Actions</string>
    </property>
    <addaction name="actionMake_Full_Connection"/>
    <addaction name="actionAlign_Vertically"/>
    <addaction name="actionAlign_Horizontally"/>
    <addaction name="actionRemove_Marks"/>
    <addaction name="actionCreate_optimal_path"/>
    <addaction name="actionNN_ABC_8_8_8"/>
    <addaction name="actionNN_GA_8_8_8"/>
   </widget>
   <addaction name="menuFile"/>
   <addaction name="menuConverter"/>
   <addaction name="menuInstruments"/>
   <addaction name="menuActions"/>
   <addaction name="menuOptions"/>
  </widget>
  <action name="actionOpen">
   <property name="text">
    <string>Open ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+O</string>
   </property>
  </action>
  <action name="actionExit">
   <property name="text">
    <string>Exit</string>
   </property>
  </action>
  <action name="actionCreateMnDataFile">
   <property name="text">
    <string>Mininet Script ...</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+M</string>
   </property>
  </action>
  <action name="actionShowPorts">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Show ports</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+P</string>
   </property>
  </action>
  <action name="actionCreateNetworkMap">
   <property name="text">
    <string>Network Map ...</string>
   </property>
  </action>
  <action name="actionSave">
   <property name="text">
    <string>Save</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+S</string>
   </property>
  </action>
  <action name="actionSaveAs">
   <property name="text">
    <string>Save as ...</string>
   </property>
  </action>
  <action name="actionDisplayDelay">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Delay</string>
   </property>
   <property name="shortcut">
    <string>Shift+D</string>
   </property>
  </action>
  <action name="actionDisplayBandwidth">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Bandwidth</string>
   </property>
   <property name="shortcut">
    <string>Shift+B</string>
   </property>
  </action>
  <action name="actionDisplayPacketLoss">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Packet Loss Rate</string>
   </property>
   <property name="shortcut">
    <string>Shift+P</string>
   </property>
  </action>
  <action name="actionHost">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Host</string>
   </property>
   <property name="shortcut">
    <string>H</string>
   </property>
  </action>
  <action name="actionSwitch">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Switch</string>
   </property>
   <property name="shortcut">
    <string>S</string>
   </property>
  </action>
  <action name="actionLink">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Link</string>
   </property>
   <property name="shortcut">
    <string>L</string>
   </property>
  </action>
  <action name="actionCreate">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Create</string>
   </property>
  </action>
  <action name="actionDelete">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Delete</string>
   </property>
  </action>
  <action name="actionMove">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Move</string>
   </property>
  </action>
  <action name="actionEdit">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>true</bool>
   </property>
   <property name="text">
    <string>Edit</string>
   </property>
   <property name="shortcut">
    <string>E</string>
   </property>
  </action>
  <action name="actionSdnController">
   <property name="checkable">
    <bool>true</bool>
   </property>
   <property name="checked">
    <bool>false</bool>
   </property>
   <property name="text">
    <string>Controller</string>
   </property>
   <property name="shortcut">
    <string>C</string>
   </property>
  </action>
  <action name="actionMake_Full_Connection">
   <property name="text">
    <string>Full Connection</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+F</string>
   </property>
  </action>
  <action name="actionNew">
   <property name="text">
    <string>New</string>
   </property>
   <property name="shortcut">
    <string>Ctrl+N</string>
   </property>
  </action>
  <action name="actionText">
   <property name="text">
    <string>Text</string>
   </property>
   <property name="shortcut">
    <string>T</string>
   </property>
  </action>
  <action name="actionWeights_Matrix">
   <property name="text">
    <string>Metric Data ...</string>
   </property>
  </action>
  <action name="actionAlign_Vertically">
   <property name="text">
    <string>Align Vertically</string>
   </property>
   <property name="shortcut">
    <string>Shift+V</string>
   </property>
  </action>
  <action name="actionAlign_Horizontally">
   <property name="text">
    <string>Align Horizontally</string>
   </property>
   <property name="shortcut">
    <string>Shift+H</string>
   </property>
  </action>
  <action name="actionRemove_Marks">
   <property name="text">
    <string>Remove Marks</string>
   </property>
  </action>
  <action name="actionCreate_optimal_path">
   <property name="text">
    <string>Create paths</string>
   </property>
  </action>
  <action name="actionNN_ABC_8_8_8">
   <property name="text">
    <string>NN+ABC+8-8-8</string>
   </property>
  </action>
  <action name="actionRandomMetric">
   <property name="text">
    <string>RandomMetric</string>
   </property>
  </action>
  <action name="actionNN_GA_8_8_8">
   <property name="text">
    <string>NN+GA+8-8-8</string>
   </property>
  </action>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>NetworkMapView</class>
   <extends>QLabel</extends>
   <header>NetworkMapView.h</header>
  </customwidget>
 </customwidgets>
 <resources>
  <include location="../images/img.qrc"/>
 </resources>
 <connections/>
</ui>
